### PR TITLE
UI for Handedness + Reset First Use.

### PR DESCRIPTION
**Pull request by andybak**
… with other buttons.

### CODE REVIEW

**Comment by andybak** [comment]
I've done a quick PCVR test. I'll test on Quest once the CI has completed.

**Comment by mikeage** [comment]
I tried this on a Q1.

Good news: it works :-)

Two thoughts though:
1. The icon for left handed mode / switch back to right handed mode is blurry; the hand on the right doesn't have distinct fingers. It's probably a scaling issue or something, because it looks perfectly sharp in the default version, but when it zooms it (since it's "raised"), it doesn't look very good. I didn't have a chance to test on a Q2, which is often sharper (or at least less subject to a screen door effect).
3. When enabling the "reset first use", the popup is right on top of my pointer hand. I tend to have that one closer to my face, and it's not readable. Should it appear on top of the palette hand, and not the brush hand? (there might be other popups that also do this, so if it's our pattern, definitely keep it)

**Review by mikeage — COMMENTED** [review]
I tried this on a Q1.

Good news: it works :-)

Two thoughts though:

1. The icon for left handed mode / switch back to right handed mode is blurry; the hand on the right doesn't have distinct fingers. It's probably a scaling issue or something, because it looks perfectly sharp in the default version, but when it zooms it (since it's "raised"), it doesn't look very good. I didn't have a chance to test on a Q2, which is often sharper (or at least less subject to a screen door effect).
2. When enabling the "reset first use", the popup is right on top of my pointer hand. I tend to have that one closer to my face, and it's not readable. Should it appear on top of the palette hand, and not the brush hand? (there might be other popups that also do this, so if it's our pattern, definitely keep it)

**Comment by andybak** [comment]
@mikeage I agree on all points but I'd probably argue in favour of getting this in and fixing the cosmetic issues later.

**Comment by andybak** [comment]
> @mikeage I agree on all points but I'd probably argue in favour of getting this in and fixing the cosmetic issues later.

Actually - I think I just resized the icons incorrectly. Gimme a moment on that point at least.

**Comment by mikeage** [comment]
> @mikeage I agree on all points but I'd probably argue in favour of getting this in and fixing the cosmetic issues later.

Sure; I'm raising them for consideration, not as blockers

File: Assets/Scripts/GUI/AppSettingsPanel.cs [hunk 1]
```diff
@@ -28,9 +28,37 @@ public override void InitPanel()
...
         }
 
+        public void HandleToggleHandedness(ToggleButton btn)
+        {
+            InputManager.m_Instance.WandOnRight = !btn.m_IsToggledOn;
+        }
+
+        public void HandleResetFirstUse()
+        {
+            if (!PanelManager.m_Instance.AdvancedModeActive())
```
Comment (mikeage) [inline Assets/Scripts/GUI/AppSettingsPanel.cs:38]: I didn't test this, but is this condition correct? Should the `!` be there?

**Comment by mikeage** [comment]
@mikeskydev , what's the latest commit for? Is that related to this PR?

**Comment by mikeskydev** [comment]
@mikeage an attempt to fix the icon aliasing via adding it to the atlas, that evolved into a tidy up with other atlased icons!

**Comment by mikeage** [comment]
Ok, cool, will check on Q1 as soon as it's done

File: Assets/Scripts/GUI/AppSettingsPanel.cs (hunk 1 above)
Comment (mikeskydev) [reply Assets/Scripts/GUI/AppSettingsPanel.cs]: Fixed this on 192301a

**Comment by mikeage** [comment]
Ok, it's no longer blurry, but still two issues:

1. The first selection doesn't work. I have to select the option twice for it to switch to left handed mode (the first seems to set it to right hand mode; is there a default missing here?). This was actually an issue before as well; I'd missed it.
2. It's no longer blurry, but at arms length, you can't see the fingers in the Q1 in right hand mode, making it... confusing?

These are three pictures showing the initial state,  after one click, and after a second click. There's a fourth picture at the bottom with an up close look at right hand mode, which is either after 1 click or 3 clicks.

WhatsApp Image 2023-02-27 at 19 17 04
WhatsApp Image 2023-02-27 at 19 17 28
WhatsApp Image 2023-02-27 at 19 17 40

Right hand mode, up close:

WhatsApp Image 2023-02-27 at 19 17 55

**Comment by andybak** [comment]
Added new icons and fixed the initial button state. How's it looking now.

**Comment by mikeage** [comment]
Turns out my headset is totally dead. Might be able to get to this in the AM

**Comment by mikeage** [comment]
Icons look great.

However, if I switch the hands using the gesture, the icon is out of sync. Upon restarting the app, the icon status is updated correctly. Merely exiting and reentering the settings screen is insufficient

**Comment by TimAidley** [comment]
I just had a go, and I like it. Looks good to me!

**Comment by mikeage** [comment]
I'm still not seeing it working. Three scenarios, all failing:
1. Open the settings screen, tap the controllers together. Button doesn't update in real time.
2. Open the settings screen, tap the controllers together. Exit settings, re-enter. Button still shows the previous state.
3. Don't have the settings screen open. Tap the controllers together. Open settings. Button still shows the previous state.

(previous state means that if we were in Right Hand mode, and tapped to move the brush to the left hand, the button still shows "Swap brush to left hand")

**Comment by mikeskydev** [comment]
Rather than storing state and having to worry about that, how about a simple 'swap hands' button with the 'refresh' looking icon, which is also visible on the sfx when you do the gesture-based switch. Maybe it also makes sense to invoke the switch via the pre-existing method with a force override, which will also hopefully get us the nice sfx.

**Comment by mikeage** [comment]
That is probably the quickest fix :-)

It's also a reasonable UX; it avoids the question of "does this button show the current state or the outcome of pushing it", and it's not like the user doesn't immediately see which had is the brush and which hand has the tools and palettes!

**Comment by andybak** [comment]
> Rather than storing state and having to worry about that, how about a simple 'swap hands' button with the 'refresh' looking icon, which is also visible on the sfx when you do the gesture-based switch. Maybe it also makes sense to invoke the switch via the pre-existing method with a force override, which will also hopefully get us the nice sfx.

Now why didn't I think of that in the first place?

Done and pushed (but not tested yet - I'll wait for the build but @mikeage feel free to jump in as our chief tester!)

**Comment by andybak** [comment]
Tested in the editor. I just kept the right-hand icon. I think it still makes sense in context.

**Review by mikeage — APPROVED** [review]
looks good!

**Comment by mikeage** [comment]
@andybak , anything else remaining, or can we merge this one?

**Comment by andybak** [comment]
Making it so

**Pull request merged**